### PR TITLE
docs: upload progress

### PR DESCRIPTION
**Pull request by trim21**
# Summary

# Checklist

- [x] I understand that this PR may be closed in case there was no previous discussion. (This doesn't apply to typos!)
- [ ] I've added a test for each change that was introduced, and I tried as much as possible to make a single atomic change.
- [ ] I've updated the documentation accordingly.

### CODE REVIEW

**Comment by Kludex** [comment]
Should we just include a blog post about it instead of including in the docs?

**Comment by trim21** [comment]
> Seems reasonable to include it in the docs seeing as we've got download tracking examples?

Why I submit this pr

File: docs/advanced.md [hunk 1]
```diff
@@ -426,6 +426,38 @@ with tempfile.NamedTemporaryFile() as download_file:
...
+                   total=p.stat().st_size) as bar:
+        with p.open("rb") as f:
+            while data := f.read(4096):
+                yield data
+                bar.update(len(data))
+
+
+file_to_upload = Path('...')
+
+url = "..."
```
[the remark this answers is not in the document]
Comment (trim21) [reply docs/advanced.md:454]: should i replace file_to_upload = Path('...') with random bytes？

File: docs/advanced.md [hunk 2]
```diff
@@ -426,6 +426,38 @@ with tempfile.NamedTemporaryFile() as download_file:
...
+
+```python
+from pathlib import Path
+
+import httpx
+from tqdm import tqdm
+
+
+def gen(p: Path):
+    with tqdm.tqdm(ascii=True, unit_scale=True, unit='B', unit_divisor=1024,
```
Comment (trim21) [inline docs/advanced.md:444]:
```suggestion
    with tqdm(ascii=True, unit_scale=True, unit='B', unit_divisor=1024,
```

File: docs/advanced.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (trim21) [reply docs/advanced.md]: I use `io.BytesIO(random.randbytes(total))` as file object to upload.

**Comment by trim21** [comment]
@tomchristie Any more suggestion?

**Review by tomchristie — APPROVED** [review]
Looks pretty good to me.
Walrus, so only works with 3.8+ but that's probs okay.
`httpbin` takes a long time to return the response from that endpoint which is a bit confusing when running the sample, but again probably fine.

**Comment by trim21** [comment]
> Walrus, so only works with 3.8+ but that's probs okay.

python 3.7 reached its EOL 2 weeks ago...

**Comment by tomchristie** [comment]
We okay with this @encode/maintainers?

(Hit merge if ya think so I guess)

**Review by Kludex — APPROVED** [review]

File: docs/advanced.md [hunk 3]
```diff
@@ -426,6 +426,39 @@ with tempfile.NamedTemporaryFile() as download_file:
 
 ![rich progress bar](img/rich-progress.gif)
 
+
```
Comment (karosis88) [inline docs/advanced.md:429]:
```suggestion

```

**Pull request merged**